drivers/s3.go

```diff
@@ -51,16 +52,31 @@ type s3Session struct {
 	fields      map[string]string
 }
 
-// S3BUCKET s3 bucket owned by this node
-var S3BUCKET string
+// S3CONFIG s3 bucket or url owned by this node
+var S3CONFIG string
 
-func s3Host(bucket string) string {
-	return fmt.Sprintf("https://%s.s3.amazonaws.com", bucket)
+// Returns host, region, bucket
+func s3Host(s3config string) (string, string, string) {
+	u, err := url.Parse(s3config)
+	// This means it's e.g. "us-west-2/my-bucket"
+	if err != nil || u.Scheme == "" {
+		br := strings.Split(s3config, "/")
+		return fmt.Sprintf("https://%s.s3.amazonaws.com", br[0]), br[0], br[1]
```

Comment:
should be 
`return fmt.Sprintf("https://%s.s3.amazonaws.com", br[1]), br[0], br[1]`

Reply:
True. What would be the most appropriate way to handle that? I could add an `err` return value, I suppose.